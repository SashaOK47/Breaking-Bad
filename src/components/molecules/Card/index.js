import style from "./styles.module.scss";

const Card = ({ card }) => {
  return (
    <div className={style.card}>
      <div className={style.photoBlock}>
        <img className={style.photo} src={card.img} alt="Photo" />
      </div>
      <div className={style.description}>
        <p className={style.status}>{card.status}</p>
        <p className={style.name}>{card.name}</p>
        <time className={style.date} dateTime="2001-11-23">
          {card.birthday}
        </time>
      </div>
    </div>
  );
};

export default Card;
