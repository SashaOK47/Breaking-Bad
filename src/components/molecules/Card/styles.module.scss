@import "../../../assets/styles/main.scss";
.card {
  max-width: 23.75rem;
  width: 100%;
  display: flex;
  background: $white;
  border-radius: 0.25rem;
  cursor: pointer;
  @include boxShadowCard;
  transition: transform 0.5s, box-shadow 0.5s;
  &:hover {
    @include boxShadowCardHover;
    transform: translateY(-0.125rem);
  }
  .photoBlock {
    width: 7.75rem;
    height: 7.75rem;
    .photo {
      width: 100%;
      height: 100%;
      object-fit: cover;
      object-position: top;
      border-radius: 0.25rem 0 0 0.25rem;
    }
  }
  .description {
    padding: 1rem 1.5rem;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    .status {
      padding: 0.5rem 0.75rem;
      margin-bottom: 0.75rem;
      background-color: $colorBgStatus;
      @include fontSmallCaps;
    }
    .date {
      @include fontSmall;
      opacity: 0.4;
    }
  }
}
