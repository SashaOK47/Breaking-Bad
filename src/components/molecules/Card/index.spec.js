import React from "react";
import Card from "./index";

const character = {
  char_id: 1,
  name: "Test",
  status: "test",
  img: "https://test.test",
  birthday: "12.12.12",
};
const grid = true;

const setUp = (props) => shallow(<Card {...props} />);

describe("Card component", () => {
  it("should render Card component", () => {
    const component = setUp({ character });
    expect(component).toMatchSnapshot();
  });

  it("class is added list", () => {
    const component = setUp({ character, grid });
    expect(component).toMatchSnapshot();
  });
});
