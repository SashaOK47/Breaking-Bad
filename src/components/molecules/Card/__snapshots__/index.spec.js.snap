// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component class is added list 1`] = `
<Link
  className="card"
  to="/character/1"
>
  <div
    className="photoBlock"
  >
    <img
      alt="Test"
      className="photo"
      src="https://test.test"
    />
  </div>
  <div
    className="description"
  >
    <p
      className="status"
    >
      test
    </p>
    <p
      className="name"
    >
      Test
    </p>
    <time
      className="date"
    >
      12.12.12
    </time>
  </div>
</Link>
`;

exports[`Card component should render Card component 1`] = `
<Link
  className="card list"
  to="/character/1"
>
  <div
    className="photoBlock"
  >
    <img
      alt="Test"
      className="photo"
      src="https://test.test"
    />
  </div>
  <div
    className="description"
  >
    <p
      className="status"
    >
      test
    </p>
    <p
      className="name"
    >
      Test
    </p>
    <time
      className="date"
    >
      12.12.12
    </time>
  </div>
</Link>
`;
